<!DOCTYPE html>
<html lang="en"><head>
<script src="slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="slides_files/libs/quarto-html/popper.min.js"></script>
<script src="slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="slides_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.2.269">

  <meta name="author" content="Clément Bernard">
  <meta name="dcterms.date" content="2023-11-27">
  <title>RNA 3D structures</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link rel="stylesheet" href="style.css">
  <link href="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">RNA 3D structures</h1>
  <p class="subtitle">M2 - GENIOMHE</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Clément Bernard 
</div>
</div>
</div>

  <p class="date">11/27/23</p>
</section>
<section>
<section id="rna-structure-description" class="title-slide slide level1 center">
<h1>RNA structure description</h1>

</section>
<section id="slides" class="slide level2">
<h2>Slides</h2>
<p>Slides are available at:</p>
<p>https://evryrna.github.io/rna_3d_bioinfo_m2_geniomhe/</p>
</section>
<section id="nucleotides" class="slide level2">
<h2>Nucleotides</h2>

<img data-src="img/nucleotide.png" style="width:50.0%" class="r-stretch quarto-figure-center"><p class="caption">Nucleotide description adapted from <a href="https://upload.wikimedia.org/wikipedia/commons/d/d3/0322_DNA_Nucleotides.jpg">Wikipedia</a></p></section>
<section id="nucleotides-chain" class="slide level2">
<h2>Nucleotides chain</h2>

<img data-src="img/rna_schema.drawio.png" style="width:90.0%" class="r-stretch quarto-figure-center"><p class="caption">Adaptation of <a href="https://www.genome.gov/genetics-glossary/RNA-Ribonucleic-Acid">article</a>: description of RNA molecular components</p></section>
<section id="rna-edges" class="slide level2">
<h2>RNA edges</h2>

<img data-src="img/rna_edges.drawio.png" class="r-stretch quarto-figure-center"><p class="caption">RNA edges where H-bonded pairs can form</p></section>
<section id="rna-edges-1" class="slide level2">
<h2>RNA edges</h2>

<img data-src="img/rna_all_edges.drawio.png" class="r-stretch quarto-figure-center"><p class="caption">All possible interactions (cis, trans, parallel, antiparallel)</p></section>
<section id="rna-base-pairs" class="slide level2">
<h2>RNA base pairs</h2>
<ul>
<li class="fragment"><p>Given orientations, the edges and base pairing, there are more than <strong>200</strong> possible base pairs.</p></li>
<li class="fragment"><p>The standard WC base pair corresponds to the cis WC/WC pairing.</p></li>
</ul>
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/wc_edge.drawio.png" style="width:50.0%"></p>
</figure>
</div>
<ul>
<li class="fragment">Non-standard interactions play a crucial role in the RNA folding process</li>
</ul>
</div>
</section>
<section id="rna-dihedral-angles" class="slide level2">
<h2>RNA dihedral angles</h2>

<img data-src="img/all_angles.drawio.png" style="width:100.0%" class="r-stretch quarto-figure-center"></section>
<section id="rna-3d-structure" class="slide level2">
<h2>RNA 3D structure</h2>

<img data-src="img/pred_rp14f_w_trans.png" class="r-stretch quarto-figure-center"><p>RNA 3D structures are complex and involve a lot of different interactions</p>
</section></section>
<section>
<section id="rna-3d-structures-prediction" class="title-slide slide level1 center">
<h1>RNA 3D structures prediction</h1>

</section>
<section id="rna-3d-structures-prediction-1" class="slide level2">
<h2>RNA 3D structures prediction</h2>
<ul>
<li class="fragment">Solved structures are stored in the <a href="https://www.rcsb.org/">PDB</a></li>
</ul>
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/pdb_all_rnas.png" style="width:80.0%"></p>
</figure>
</div>
</div>
</section>
<section id="rna-3d-structures-prediction-2" class="slide level2">
<h2>RNA 3D structures prediction</h2>

<img data-src="img/pdb_one_rna.png" class="r-stretch quarto-figure-center"><p class="caption">Example of RNA page on PDB</p></section>
<section id="rna-3d-structures-prediction-3" class="slide level2">
<h2>RNA 3D structures prediction</h2>

<img data-src="img/pdb_article.png" class="r-stretch quarto-figure-center"><p class="caption">Each RNA 3D structure is the result of hard work</p></section></section>
<section>
<section id="experimental-methods" class="title-slide slide level1 center">
<h1>Experimental methods</h1>

</section>
<section id="experimental-methods-1" class="slide level2">
<h2>Experimental methods</h2>
<h3 id="macromolecular-x-ray-crystallography-mr">Macromolecular X-ray crystallography (MR)</h3>
<ul>
<li class="fragment">Most widely used technique</li>
<li class="fragment">A crystal is exposed to an X-ray beam.</li>
<li class="fragment">Diffraction images are collected with rotation of the crystal.</li>
<li class="fragment">Application of inverse Fourier to find strucutre</li>
</ul>
</section>
<section id="experimental-methods-2" class="slide level2">
<h2>Experimental methods</h2>
<h3 id="nuclear-magnetic-resonance-nmr">Nuclear magnetic resonance (NMR)</h3>
<ul>
<li class="fragment">Observe local magnetic fields around atom and measure interactions between them</li>
<li class="fragment">Atoms carry magnetic that take various orientations characterized by different energies when applied to magnetic field.</li>
</ul>
</section>
<section id="experimental-methods-3" class="slide level2">
<h2>Experimental methods</h2>
<h3 id="electron-microscopy-em">Electron microscopy (EM)</h3>
<ul>
<li class="fragment">Rely on the constrast formed by the interaction of incoming electrons</li>
<li class="fragment">Multiple 2D projection images of randomly oriented molecules are recorded</li>
<li class="fragment">3D density map is computationally combined</li>
</ul>
</section>
<section id="experimental-methods-4" class="slide level2">
<h2>Experimental methods</h2>

<img data-src="img/experimental_methods.png" class="r-stretch quarto-figure-center"></section>
<section id="experimental-methods-5" class="slide level2">
<h2>Experimental methods</h2>
<ul>
<li class="fragment"><p>Experimental methods exist but are laborious and of high cost (in <strong>time</strong> and <strong>money</strong>)</p></li>
<li class="fragment"><p>Computational methods aim to predict RNA 3D structures with high throughput</p></li>
</ul>
<div class="fragment">
<p><strong>Question</strong>: How to compute RNA 3D structures ?</p>
</div>
</section></section>
<section>
<section id="rna-3d-structures-prediction-4" class="title-slide slide level1 center">
<h1>RNA 3D structures prediction</h1>

</section>
<section id="rna-3d-structures-prediction-5" class="slide level2">
<h2>RNA 3D structures prediction</h2>
<h3 id="general-pipeline">General pipeline</h3>
<p>Models that predict RNA 3D structures use this general workflow:</p>
<ul>
<li class="fragment">Generate RNA 3D candidates</li>
<li class="fragment">Selection of good candidates (structure evaluation)</li>
<li class="fragment">Refinement: relaxe the structures</li>
<li class="fragment">Final prediction: return either the best structure or multiple structures</li>
</ul>
</section>
<section id="rna-3d-structures-prediction-6" class="slide level2">
<h2>RNA 3D structures prediction</h2>
<p>Methods can be divided into 3 classes:</p>
<ul>
<li class="fragment"><strong><em>Ab initio</em></strong> (physics-based): use of thermodynamic and physics</li>
<li class="fragment"><strong>Template-based</strong>: use of fragment assembly</li>
<li class="fragment"><strong>Deep learning</strong></li>
</ul>
</section></section>
<section>
<section id="ab-initio" class="title-slide slide level1 center">
<h1>Ab initio</h1>

</section>
<section id="ab-initio-1" class="slide level2">
<h2>Ab initio</h2>
<ul>
<li class="fragment">Rely on the fact that native structures are those states with the lowest energies for RNAs</li>
<li class="fragment">The energy used usually consider <strong>dihedral angles</strong>, <strong>bond length</strong>, <strong>base pairing</strong>, …</li>
<li class="fragment">Use a <strong>generation</strong> process and a <strong>discriminator</strong></li>
<li class="fragment">Use Monte Carlo or molecular dynamics (MD) sampling procedures</li>
</ul>
</section>
<section id="ab-initio-2" class="slide level2">
<h2>Ab initio</h2>
<h3 id="coarse-grained">Coarse-grained</h3>
<ul>
<li class="fragment">All-atom simulation can be time-consuming</li>
<li class="fragment">Use of coarse-grained to reduce computation time</li>
</ul>
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/cg.png" style="width:50.0%"></p>
</figure>
</div>
</div>
</section>
<section id="ab-initio-3" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-granularity">RNAJP granularity</h3>
<ul>
<li class="fragment">5 pseudoatoms (beads) to represent a nucleotide</li>
</ul>
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/rnajp_nucleotides.png" style="width:40.0%"></p>
</figure>
</div>
</div>
</section>
<section id="ab-initio-4" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-granularity-1">RNAJP granularity</h3>
<ul>
<li class="fragment">Use the 2D structure to represent CG helix as A-form helix</li>
</ul>
<div class="fragment">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/rnajp_helix.png" style="width:40.0%"></p>
</figure>
</div>
</div>
</section>
<section id="ab-initio-5" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-force-field">RNAJP force-field</h3>
<ul>
<li class="fragment">Force-field uses 12 terms: <span class="math display">\[
E_{total} = E_{bond} + E_{angle} + E_{torsion} + \\
          E_{base-pairing} + E_{base-stacking} + \\
          E_{excluded-volume} + \\
          E_{A helix} + E_{helix-helix}+ E_{junction-compact} + \\
          E_{JAR3D} + E_{loop-loop} + \\
          E_{PK stacking}
\]</span></li>
</ul>
</section>
<section id="ab-initio-6" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-force-field-1">RNAJP force-field</h3>
<ul>
<li class="fragment"><span class="math inline">\(E_{bond} + E_{angle} + E_{torsion}\)</span>: bonded interactions in terms of bond length, angle and torsion</li>
<li class="fragment"><span class="math inline">\(E_{base-pairing} + E_{base-stacking}\)</span>: important base pairing and base stacking interactions</li>
<li class="fragment"><span class="math inline">\(E_{excluded-volume}\)</span>: avoid steric clashes between beads</li>
<li class="fragment"><span class="math inline">\(E_{A helix} + E_{helix-helix}+ E_{junction-compact}\)</span>: use to accelerate the conformational sampling</li>
</ul>
</section>
<section id="ab-initio-7" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-force-field-2">RNAJP force-field</h3>
<ul>
<li class="fragment"><span class="math inline">\(E_{JAR3D} + E_{loop-loop}\)</span>: describe the intraloop and interloop interactions</li>
<li class="fragment"><span class="math inline">\(E_{PK stacking}\)</span>: describe the stacking interaction between the pseudoknotted (PK) helix and its adjacent non-pseudoknotted (nonPK) helix</li>
</ul>
</section>
<section id="ab-initio-8" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-sampling">RNAJP sampling</h3>
<ul>
<li class="fragment">Sampling procedure: molecular dynamics (MD)</li>
<li class="fragment">Procedure:
<ul>
<li class="fragment">Generate an initial 3D structure</li>
<li class="fragment">Fold the helices in the given 2D structure</li>
<li class="fragment">Fold the hairpin and internal loop structures</li>
<li class="fragment">Sample different junction topologies and junction loop structures</li>
</ul></li>
</ul>
</section>
<section id="ab-initio-9" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-model-selection">RNAJP: model selection</h3>
<ul>
<li class="fragment">Select the lowest-energy structure as the first predicted model</li>
<li class="fragment">Among a pool of top 1% low energy, select the structure with the largest RMSD from previous selected structure</li>
<li class="fragment">Select the structure among the top 1% low-energy structures with the largest sum of RMSDs relative to the first and second models as the third model</li>
<li class="fragment">Continue until N predictions</li>
</ul>
</section>
<section id="ab-initio-10" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-all-atom-reconstruction">RNAJP: all-atom reconstruction</h3>
<ul>
<li class="fragment">CG beads aligned in the standard base</li>
<li class="fragment">Selection of reconstructed atoms after superimposition</li>
</ul>
</section>
<section id="ab-initio-11" class="slide level2">
<h2>Ab initio</h2>
<h3 id="rnajp-refinement">RNAJP: refinement</h3>
<ul>
<li class="fragment">Fix the broken bonds and remove steric clashes using QRNAS (<span class="citation" data-cites="qrnas">Stasiewicz et al. (<a href="#/references" role="doc-biblioref" onclick="">2019</a>)</span>)</li>
</ul>
</section>
<section id="ab-initio-12" class="slide level2">
<h2>Ab initio</h2>
<h3 id="limits">Limits</h3>
<ul>
<li class="fragment">Limited by the sequence length (increase computation time)</li>
<li class="fragment">Require efficient sampling function (to avoid useless computation)</li>
<li class="fragment">Methods usually slow</li>
</ul>
</section></section>
<section>
<section id="template-based" class="title-slide slide level1 center">
<h1>Template-based</h1>

</section>
<section id="template-based-1" class="slide level2">
<h2>Template-based</h2>
<ul>
<li class="fragment">Creation of 3D structures from known structures (templates)</li>
<li class="fragment">Use of fragment mapping between 2D and 3D</li>
<li class="fragment">Align the target sequence to the template</li>
<li class="fragment">Structure evaluation to choose the best candidate</li>
</ul>
</section>
<section id="template-based-2" class="slide level2">
<h2>Template-based</h2>
<h3 id="granularity">Granularity</h3>
<ul>
<li class="fragment">Looking to template at the nucleotide level is time consuming</li>
<li class="fragment">Sample the space not at the atom level but at the level of basic block</li>
</ul>
</section>
<section id="template-based-3" class="slide level2">
<h2>Template-based</h2>
<h3 id="granularity-1">Granularity</h3>
<ul>
<li class="fragment">Basic blocks can be:
<ul>
<li class="fragment">Three-nucleotides (used in FARNA) <img data-src="img/tp_3nt.png" style="width:50.0%" data-fig-align="center"></li>
</ul></li>
</ul>
</section>
<section id="template-based-4" class="slide level2" data-transition="none">
<h2>Template-based</h2>
<h3 id="granularity-2">Granularity</h3>
<div>
<ul>
<li>Basic blocks can be:
<ul>
<li>Three-nucleotides (used in FARNA)</li>
<li>Smallest Secondary Elements (SSEs)</li>
</ul></li>
</ul>
</div>

<img data-src="img/tp_sses.png" style="width:50.0%" class="r-stretch quarto-figure-center"></section>
<section id="template-based-5" class="slide level2">
<h2>Template-based</h2>
<h3 id="example-3drna">Example: 3dRNA</h3>

<img data-src="img/3drna_workflow.png" class="r-stretch quarto-figure-center"></section>
<section id="template-based-6" class="slide level2">
<h2>Template-based</h2>
<h3 id="example-3drna-1">Example: 3dRNA</h3>
<ul>
<li class="fragment">Assembled structures are clustered by K-means</li>
<li class="fragment">Use RMSD values to do the clustering</li>
<li class="fragment">Centroids of each cluster are evaluated by 3dRNAscore</li>
</ul>
</section>
<section id="template-based-7" class="slide level2">
<h2>Template-based</h2>
<h3 id="limits-1">Limits</h3>
<ul>
<li class="fragment">Rely on representative templates</li>
<li class="fragment">Fail to generalise to unseen structures</li>
<li class="fragment">Limited by the number of available RNA</li>
</ul>
</section></section>
<section>
<section id="towards-deep-learning" class="title-slide slide level1 center">
<h1>Towards deep-learning</h1>

</section>
<section id="towards-deep-learning-1" class="slide level2">
<h2>Towards deep-learning</h2>
<h3 id="proteins">Proteins</h3>
<ul>
<li class="fragment">AlphaFold 2(Jumper et al., 2021) has successfully predicted protein 3D structures</li>
</ul>
<div class="fragment">
<p><img data-src="img/alphafold.png"></p>
</div>
</section>
<section id="towards-deep-learning-2" class="slide level2">
<h2>Towards deep-learning</h2>
<h3 id="proteins-1">Proteins</h3>
<p>AlphaFold not adapted to RNAs:</p>
<ul>
<li class="fragment">RNAs composed of nucleotides (compared to amino acid)</li>
<li class="fragment">Disparity in length for RNAs</li>
<li class="fragment">Multiple structures possible for RNAs</li>
<li class="fragment">Fewer structures available</li>
<li class="fragment">AlphaFold is based on Multiple Sequence Alignment (MSA), but not all RNA families are known</li>
</ul>
</section>
<section id="deep-learning" class="slide level2">
<h2>Deep learning</h2>
<ul>
<li class="fragment">Recent works try to reproduce AlphaFold</li>
<li class="fragment">Remain limited: works not published (only preprint)</li>
</ul>
<div class="r-stack">
<p><img data-src="img/preprint_0.png" class="absolute fragment" style="top: 350px; left: 100px; width: 450px; height: 300px; "> <img data-src="img/preprint_1.png" class="absolute fragment" style="top: 300px; left: 120px; width: 850px; height: 300px; " data-rigth="20"> <img data-src="img/preprint_2.png" class="absolute fragment" style="top: 350px; width: 450px; height: 300px; " data-rigth="20"> <img data-src="img/preprint_3.png" class="absolute fragment" style="top: 350px; left: 80px; width: 450px; height: 300px; " data-rigth="20"></p>
</div>
</section>
<section id="deep-learning-1" class="slide level2">
<h2>Deep learning</h2>
<h3 id="example-drfold">Example: DRfold</h3>

<img data-src="img/preprint_3.png" class="r-stretch quarto-figure-center"><ul>
<li class="fragment">Example of work published that had good performances on a recent challenge</li>
</ul>
</section>
<section id="deep-learning-2" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-atom-representation">DRfold atom representation</h3>

<img data-src="img/drfold_cg.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-3" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-architecture">DRfold architecture</h3>

<img data-src="img/drfold_architecture.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-4" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-embedding">DRfold embedding</h3>

<img data-src="img/drfold_emb.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-5" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-transformer">DRfold transformer</h3>

<img data-src="img/drfold_transformer.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-6" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-structure">DRfold structure</h3>

<img data-src="img/drfold_structure.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-7" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-geometry">DRfold geometry</h3>

<img data-src="img/drfold_geometry.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-8" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-fape">DRfold FAPE</h3>

<img data-src="img/drfold_fape.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-9" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-loss">DRfold loss</h3>
<p><span class="math display">\[
L_{e2e} = 1.5 L_{FAPE} + 0.6L_{dist}
\]</span></p>
<p>with <span class="math display">\[
L_{dist} = - \sum_{i,j} \sum_{b=1}^{36} y_{ij}^b \log p_{ij}^b
\]</span></p>
<p><span class="math inline">\(y_{ij}^b\)</span> assesses if the distance of atom pair <span class="math inline">\((i, j)\)</span> in the target experimental structure falls into the <span class="math inline">\(b\)</span>-th distance</p>
</section>
<section id="deep-learning-10" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-dataset">DRfold: dataset</h3>
<ul>
<li class="fragment">Training set of 3864 unique RNAs extracted from the PDB</li>
<li class="fragment">Test set on 40 non-redundant RNAs</li>
<li class="fragment">RNA length from 14 to 392 nucleotides</li>
</ul>
</section>
<section id="deep-learning-11" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold-results">DRfold: results</h3>

<img data-src="img/drfold_results.png" class="r-stretch quarto-figure-center"></section>
<section id="deep-learning-12" class="slide level2">
<h2>Deep learning</h2>
<h3 id="drfold">DRfold</h3>
<ul>
<li class="fragment">Use of full-atom reconstruction</li>
<li class="fragment">Training took 2 weeks on Nvidia A40 GPUwith 32GB of memory</li>
<li class="fragment">RNA sequences of 200 nucleotides maximum</li>
<li class="fragment"><strong>Doesn’t use MSA nor self-distillation</strong></li>
</ul>
</section>
<section id="deep-learning-13" class="slide level2">
<h2>Deep learning</h2>
<h3 id="limits-2">Limits</h3>
<ul>
<li class="fragment">Overall performance of deep learning approaches still to be improved</li>
<li class="fragment">Lack of sufficient solved structures</li>
<li class="fragment">Interpretation of the model</li>
<li class="fragment">Integration of environment conditions (ions ?)</li>
</ul>
</section></section>
<section>
<section id="rna-quality-measurement" class="title-slide slide level1 center">
<h1>RNA quality measurement</h1>

</section>
<section id="rna-quality-measurement-1" class="slide level2">
<h2>RNA quality measurement</h2>

<img data-src="img/1d_2d_3d_evaluation.drawio.png" style="width:200.0%" class="r-stretch"><ul>
<li class="fragment">Evaluation of raw sequence or 2D structure is accessible (F1-score, MCC, …)</li>
<li class="fragment">RNA 3D structures are complex objects: hard to evaluate (pairing, geometries, etc)</li>
</ul>
</section>
<section id="rna-quality-measurement-2" class="slide level2">
<h2>RNA quality measurement</h2>

<img data-src="img/rp14f_pred.drawio.png" style="width:20.0%" class="r-stretch quarto-figure-center"><p><strong>Given a structure, how to assess its quality ?</strong></p>
</section>
<section id="metrics" class="slide level2">
<h2>Metrics</h2>
<h3 id="definition">Definition</h3>
<div class="columns">
<div class="column" style="width:50%;">
<h4 id="predicted-structure">Predicted structure</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/rp14f_pred.drawio.png" style="width:50.0%"></p>
</figure>
</div>
</div><div class="column" style="width:50%;">
<h4 id="reference-structure">Reference structure</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/ref_rp14f.drawio.png" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<p><strong>Metrics</strong>: Quality measurement given a solved structure</p>
</section>
<section id="metrics-1" class="slide level2">
<h2>Metrics</h2>
<h3 id="general-metrics">General metrics</h3>
<ul>
<li class="fragment"><strong>RMSD</strong>: Atom deviation</li>
<li class="fragment"><strong>CLASH</strong> (<span class="citation" data-cites="molprobity">Davis et al. (<a href="#/references" role="doc-biblioref" onclick="">2007</a>)</span>): Steric clashes<br>
</li>
<li class="fragment"><strong><span class="math inline">\(\epsilon\)</span>RMSD</strong> (<span class="citation" data-cites="barnaba">Bottaro, Di Palma, and Bussi (<a href="#/references" role="doc-biblioref" onclick="">2014</a>)</span>): Relative distance</li>
</ul>
</section>
<section id="metrics-2" class="slide level2">
<h2>Metrics</h2>
<h3 id="protein-based">Protein-based</h3>
<ul>
<li class="fragment"><strong>GDT-TS</strong> (<span class="citation" data-cites="gdt">Zemla et al. (<a href="#/references" role="doc-biblioref" onclick="">1999</a>)</span>): Superimposed residues</li>
<li class="fragment"><strong>TM-score</strong> (<span class="citation" data-cites="tm-score-rna">Y. Zhang and Skolnick (<a href="#/references" role="doc-biblioref" onclick="">2004</a>)</span>): Largest deviation</li>
<li class="fragment"><strong>lDDT</strong> (<span class="citation" data-cites="lddt">Mariani et al. (<a href="#/references" role="doc-biblioref" onclick="">2013</a>)</span>): Interatomic distance</li>
<li class="fragment"><strong>CAD-score</strong> (<span class="citation" data-cites="cad-score">Olechnovič, Kulberkytė, and Venclovas (<a href="#/references" role="doc-biblioref" onclick="">2013</a>)</span>): Contact-area</li>
</ul>
</section>
<section id="metrics-3" class="slide level2">
<h2>Metrics</h2>
<h3 id="rna-oriented">RNA-oriented</h3>
<ul>
<li class="fragment"><strong>INF, DI</strong> (<span class="citation" data-cites="inf">Parisien et al. (<a href="#/references" role="doc-biblioref" onclick="">2009</a>)</span>): Key interactions accuracy</li>
<li class="fragment"><strong>P-VALUE</strong> (<span class="citation" data-cites="p-value">Hajdin et al. (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span>): Non-randomness</li>
<li class="fragment"><strong>MCQ</strong> (<span class="citation" data-cites="mcq">Zok, Popenda, and Szachniuk (<a href="#/references" role="doc-biblioref" onclick="">2013</a>)</span>): Angle deviations</li>
</ul>
</section>
<section id="metrics-4" class="slide level2">
<h2>Metrics</h2>
<h3 id="example-interaction-network-fidelity">Example: Interaction Network Fidelity</h3>
<p><span class="math display">\[
INF = \sqrt{( \frac{TP}{TP+FP}) \times (\frac{TP}{TP + FN})}
\]</span> with TP the true positive of the considered interactions</p>
<ul>
<li class="fragment">When the model reproduces exactly the reference interactions, |TP| &gt; 1 and FP = FN = 0. -&gt; The INF value equals 1</li>
</ul>
</section>
<section id="metrics-5" class="slide level2">
<h2>Metrics</h2>
<h3 id="example-interaction-network-fidelity-1">Example: Interaction Network Fidelity</h3>
<ul>
<li class="fragment"><p>If none of the interactions of the reference is reproduced in the predicted model, |TP| = 0, and the INF score equals 0.</p></li>
<li class="fragment"><p>INF score can be specific to base-pairing interactions (<span class="math inline">\(INF_{bp}\)</span>), the base stacking interactions (<span class="math inline">\(INF_{stack}\)</span>), or consider both (<span class="math inline">\(INF_{all}\)</span> )</p></li>
</ul>
</section>
<section id="metrics-6" class="slide level2">
<h2>Metrics</h2>
<ul>
<li class="fragment"><p>Metrics help to evaluate a prediction given a solved structure</p></li>
<li class="fragment"><p>No universal metric exists: RMSD is mainly used (but sensible to outliers)</p></li>
<li class="fragment"><p><strong>Metrics requires having a solved structure, which is not available when model development</strong></p></li>
</ul>
</section></section>
<section>
<section id="scoring-functions" class="title-slide slide level1 center">
<h1>Scoring functions</h1>

</section>
<section id="scoring-functions-1" class="slide level2">
<h2>Scoring functions</h2>

<img data-src="img/scoring_fig.drawio.png" style="width:105.0%" class="r-stretch quarto-figure-center"><ul>
<li class="fragment">When developing models, a lot of structures can be generated and should be ranked</li>
<li class="fragment">Ranking can be used to return multiple structures</li>
</ul>
<div class="fragment">
<p><strong>Scoring function</strong>: predict the quality of a structure <strong>without a known reference</strong></p>
</div>
</section>
<section id="scoring-functions-2" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="knowledged-based">Knowledged based</h3>
<p>Scoring function based on knowledge-based statistical potentials</p>
<ul>
<li class="fragment"><strong>RASP</strong> (<span class="citation" data-cites="rasp">Capriotti et al. (<a href="#/references" role="doc-biblioref" onclick="">2011</a>)</span>): Pairwise-distances</li>
<li class="fragment"><strong><span class="math inline">\(\epsilon\)</span>SCORE</strong> (<span class="citation" data-cites="barnaba">Bottaro, Di Palma, and Bussi (<a href="#/references" role="doc-biblioref" onclick="">2014</a>)</span>): Relative positions</li>
<li class="fragment"><strong>3dRNASCORE</strong> (<span class="citation" data-cites="3drnascore">Wang et al. (<a href="#/references" role="doc-biblioref" onclick="">2015</a>)</span>): Distance/dihedral angles</li>
<li class="fragment"><strong>DFIRE-RNA</strong> (<span class="citation" data-cites="dfire-rna">T. Zhang et al. (<a href="#/references" role="doc-biblioref" onclick="">2019</a>)</span>): Pairwise-distances</li>
<li class="fragment"><strong>rsRNASP</strong> (<span class="citation" data-cites="rsrnasp">Tan et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span>): Short/long pairwise distances</li>
</ul>
</section>
<section id="scoring-functions-3" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="deep-learning-based">Deep learning based</h3>
<p>Scoring functions based on deep learning architectures</p>
<ul>
<li class="fragment"><strong>RNA3DCNN</strong> (<span class="citation" data-cites="rna3dcnn">Li (<a href="#/references" role="doc-biblioref" onclick="">2018</a>)</span>): Atom grid</li>
<li class="fragment"><strong>ARES</strong> (<span class="citation" data-cites="ares">Townshend et al. (<a href="#/references" role="doc-biblioref" onclick="">2021</a>)</span>): Atom position/chemical type</li>
</ul>
</section>
<section id="scoring-functions-4" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="example-rnad3dcnn">Example: RNAD3DCNN</h3>
<ul>
<li class="fragment">Predict for each nucleotide an unfitness score</li>
<li class="fragment">The unfitness is based on the RMSD of its surroundings and its RMSD</li>
<li class="fragment">A score of 0 means the nucleotide fits into its environment perfectly (native conformation)</li>
<li class="fragment">For global quality assessment, the unfitness scores of all nucleotides are accumulated</li>
</ul>
</section>
<section id="scoring-functions-5" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn-inputs">RNAD3DCNN: inputs</h3>
<ul>
<li class="fragment">For each nucleotide, a box of 3 channels represents:
<ul>
<li class="fragment">Atomic occupation number</li>
<li class="fragment">Mass</li>
<li class="fragment">Charge within the grid box</li>
</ul></li>
<li class="fragment">A box is of size 32x32x32</li>
<li class="fragment">The RGB channels are mapped to the three channels</li>
</ul>
</section>
<section id="scoring-functions-6" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn-inputs-1">RNAD3DCNN: inputs</h3>

<img data-src="img/rna3dcnn_inputs.png" class="r-stretch quarto-figure-center"></section>
<section id="scoring-functions-7" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn-architecture">RNAD3DCNN: architecture</h3>

<img data-src="img/rna3dcnn_architecture.png" class="r-stretch quarto-figure-center"></section>
<section id="scoring-functions-8" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn-dataset">RNAD3DCNN: dataset</h3>
<ul>
<li class="fragment">Use of 619 non-redundant RNAs from NDB website</li>
<li class="fragment">Training done on two different datasets:
<ul>
<li class="fragment"><strong>MD dataset</strong>: a simulated annealing molecular dynamics simulation was done to each RNA</li>
<li class="fragment"><strong>MC dataset</strong>: fragment assembly method, MC process guided by a low-resolution knowledge-based energy function</li>
</ul></li>
<li class="fragment">1 million training samples and 0.2 million validation for each subdataset</li>
</ul>
</section>
<section id="scoring-functions-9" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn-results">RNAD3DCNN: results</h3>

<img data-src="img/rna3dcnn_results.png" class="r-stretch quarto-figure-center"><p class="caption">Relationship between RMSD and RNA3DCNN for an RNA</p></section>
<section id="scoring-functions-10" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn-results-1">RNAD3DCNN: results</h3>

<img data-src="img/rna3dcnn_testset.png" class="r-stretch quarto-figure-center"><p class="caption">Number of native structures with minimum scores</p></section>
<section id="scoring-functions-11" class="slide level2">
<h2>Scoring functions</h2>
<h3 id="rnad3dcnn">RNAD3DCNN</h3>
<ul>
<li class="fragment">Try to reproduce RMSD with deep learning approaches</li>
<li class="fragment">Inspired by CNN image-based models</li>
</ul>
<h3 id="limits-3">Limits</h3>
<ul>
<li class="fragment">Deep learning limits: require huge dataset, black box</li>
<li class="fragment">Training step biased by the sampling procedures</li>
<li class="fragment">Performances remain limited to novel approaches</li>
</ul>
</section>
<section id="questions" class="slide level2">
<h2>Questions ?</h2>
<p>Contact: clement.bernard@univ-evry.fr</p>
<p>Come to see my poster on Monday 4th, December</p>

<img data-src="img/rnadvisor.png" class="r-stretch quarto-figure-center"></section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>
<!-- ## Project proposal

Torsion angle definition
Link between sequence and torsion angle: SPOT-RNA
Torsion angle distribution
Project: predict torsional discretized space from the sequence -->

<img src="https://www.ibisc.univ-evry.fr/wp-content/uploads/2020/07/logo_ibiscsaclay.png" class="slide-logo r-stretch"><div class="footer footer-default">
<p><a href="">RNA 3D structures comprehension</a></p>
</div>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-barnaba" class="csl-entry" role="doc-biblioentry">
Bottaro, Sandro, Francesco Di Palma, and Giovanni Bussi. 2014. <span>“<span class="nocase">The Role of Nucleobase Interactions in RNA Structure and Dynamics</span>.”</span> <em>Nucleic Acids Research</em> 42 (October). <a href="https://doi.org/10.1093/nar/gku972">https://doi.org/10.1093/nar/gku972</a>.
</div>
<div id="ref-rasp" class="csl-entry" role="doc-biblioentry">
Capriotti, Emidio, Tomas Norambuena, Marc A. Marti-Renom, and Francisco Melo. 2011. <span>“<span class="nocase">All-atom knowledge-based potential for RNA structure prediction and assessment</span>.”</span> <em>Bioinformatics</em> 27 (April): 1086–93. <a href="https://doi.org/10.1093/bioinformatics/btr093">https://doi.org/10.1093/bioinformatics/btr093</a>.
</div>
<div id="ref-molprobity" class="csl-entry" role="doc-biblioentry">
Davis, Ian W., Andrew Leaver-Fay, Vincent B. Chen, Jeremy N. Block, Gary J. Kapral, Xueyi Wang, Laura W. Murray, et al. 2007. <span>“MolProbity: All-Atom Contacts and Structure Validation for Proteins and Nucleic Acids.”</span> <em>Nucleic Acids Research</em> 35: W375–83. https://doi.org/<a href="https://doi.org/10.1093/nar/gkm216">https://doi.org/10.1093/nar/gkm216</a>.
</div>
<div id="ref-p-value" class="csl-entry" role="doc-biblioentry">
Hajdin, Christine, Feng Ding, Nikolay Dokholyan, and Kevin Weeks. 2010. <span>“<span class="nocase">On the significance of an RNA tertiary structure prediction</span>.”</span> <em>RNA (New York, N.Y.)</em> 16 (July): 1340–49. <a href="https://doi.org/10.1261/rna.1837410">https://doi.org/10.1261/rna.1837410</a>.
</div>
<div id="ref-rna3dcnn" class="csl-entry" role="doc-biblioentry">
Li, Wei AND Wang, Jun AND Zhu. 2018. <span>“<span class="nocase">RNA3DCNN: Local and global quality assessments of RNA 3D structures using 3D deep convolutional neural networks</span>.”</span> <em>PLOS Computational Biology</em> 14 (11): 1–18. <a href="https://doi.org/10.1371/journal.pcbi.1006514">https://doi.org/10.1371/journal.pcbi.1006514</a>.
</div>
<div id="ref-lddt" class="csl-entry" role="doc-biblioentry">
Mariani, Valerio, Marco Biasini, Alessandro Barbato, and Torsten Schwede. 2013. <span>“lDDT: A Local Superposition-Free Score for Comparing Protein Structures and Models Using Distance Difference Tests.”</span> <em>Bioinformatics (Oxford, England)</em> 29 (21): 2722–28. <a href="https://doi.org/10.1093/bioinformatics/btt473">https://doi.org/10.1093/bioinformatics/btt473</a>.
</div>
<div id="ref-cad-score" class="csl-entry" role="doc-biblioentry">
Olechnovič, Kliment, Eleonora Kulberkytė, and Česlovas Venclovas. 2013. <span>“<span class="nocase">CAD-score: A new contact area difference-based function for evaluation of protein structural models</span>.”</span> <em>Proteins</em> 81 (January). <a href="https://doi.org/10.1002/prot.24172">https://doi.org/10.1002/prot.24172</a>.
</div>
<div id="ref-inf" class="csl-entry" role="doc-biblioentry">
Parisien, Marc, José Cruz, Eric Westhof, and François Major. 2009. <span>“<span class="nocase">New metrics for comparing and assessing discrepancies between RNA 3D structures and models</span>.”</span> <em>RNA (New York, N.Y.)</em> 15 (September): 1875–85. <a href="https://doi.org/10.1261/rna.1700409">https://doi.org/10.1261/rna.1700409</a>.
</div>
<div id="ref-qrnas" class="csl-entry" role="doc-biblioentry">
Stasiewicz, Juliusz, Shubham Mukherjee, Chandran Nithin, and Janusz M Bujnicki. 2019. <span>“QRNAS: Software Tool for Refinement of Nucleic Acid Structures.”</span> <em>BMC Structural Biology</em> 19 (1): 5. <a href="https://doi.org/10.1186/s12900-019-0103-1">https://doi.org/10.1186/s12900-019-0103-1</a>.
</div>
<div id="ref-rsrnasp" class="csl-entry" role="doc-biblioentry">
Tan, Ya-Lan, Xunxun Wang, Ya-Zhou Shi, Wenbing Zhang, and Zhi-Jie Tan. 2022. <span>“<span class="nocase">rsRNASP: A residue-separation-based statistical potential for RNA 3D structure evaluation</span>.”</span> <em>Biophysical Journal</em> 121 (January): 142–56. <a href="https://doi.org/10.1016/j.bpj.2021.11.016">https://doi.org/10.1016/j.bpj.2021.11.016</a>.
</div>
<div id="ref-ares" class="csl-entry" role="doc-biblioentry">
Townshend, Raphael J. L., Stephan Eismann, Andrew M. Watkins, Ramya Rangan, Maria Karelina, Rhiju Das, and Ron O. Dror. 2021. <span>“<span class="nocase">Geometric deep learning of RNA structure</span>.”</span> <em>Science</em> 373 (August): 1047–51. <a href="https://doi.org/10.1126/science.abe5650">https://doi.org/10.1126/science.abe5650</a>.
</div>
<div id="ref-3drnascore" class="csl-entry" role="doc-biblioentry">
Wang, Jian, Yunjie Zhao, Chunyan Zhu, and Yi Xiao. 2015. <span>“<span class="nocase">3dRNAscore: a distance and torsion angle dependent evaluation function of 3D RNA structures</span>.”</span> <em>Nucleic Acids Research</em> 43 (May): e63–63. <a href="https://doi.org/10.1093/nar/gkv141">https://doi.org/10.1093/nar/gkv141</a>.
</div>
<div id="ref-gdt" class="csl-entry" role="doc-biblioentry">
Zemla, Adam, Česlovas Venclovas, John Moult, and Krzysztof Fidelis. 1999. <span>“<span class="nocase">Processing and analysis of CASP3 protein structure predictions</span>.”</span> <em>Proteins: Structure, Function, and Bioinformatics</em> 37 (S3): 22–29. https://doi.org/<a href="https://doi.org/10.1002/(SICI)1097-0134(1999)37:3+<22::AID-PROT5>3.0.CO;2-W">https://doi.org/10.1002/(SICI)1097-0134(1999)37:3+&lt;22::AID-PROT5&gt;3.0.CO;2-W</a>.
</div>
<div id="ref-dfire-rna" class="csl-entry" role="doc-biblioentry">
Zhang, Tongchuan, Guodong Hu, Yuedong Yang, Jihua Wang, and Yaoqi Zhou. 2019. <span>“All-Atom Knowledge-Based Potential for RNA Structure Discrimination Based on the Distance-Scaled Finite Ideal-Gas Reference State.”</span> <em>Journal of Computational Biology : A Journal of Computational Molecular Cell Biology</em> 27 (October). <a href="https://doi.org/10.1089/cmb.2019.0251">https://doi.org/10.1089/cmb.2019.0251</a>.
</div>
<div id="ref-tm-score-rna" class="csl-entry" role="doc-biblioentry">
Zhang, Yang, and Jeffrey Skolnick. 2004. <span>“<span class="nocase">Scoring function for automated assessment of protein structure template quality</span>.”</span> <em>Proteins</em> 57 (4): 702—710. <a href="https://doi.org/10.1002/prot.20264">https://doi.org/10.1002/prot.20264</a>.
</div>
<div id="ref-mcq" class="csl-entry" role="doc-biblioentry">
Zok, Tomasz, Mariusz Popenda, and Marta Szachniuk. 2013. <span>“<span class="nocase">MCQ4Structures to compute similarity of molecule structures</span>.”</span> <em>Central European Journal of Operations Research</em> 22 (April). <a href="https://doi.org/10.1007/s10100-013-0296-5">https://doi.org/10.1007/s10100-013-0296-5</a>.
</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'slide',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'fast',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>