---
title: "RNA 3D structures"
subtitle: "M2 - GENIOMHE"
author: Clément Bernard
date: today
format:
  revealjs: 
    # theme: serif
    theme: [default, custom.scss]
    transition: slide
    transition-speed: fast
    slide-number: c/t
footer: "[RNA 3D structures comprehension]()"
logo: https://www.ibisc.univ-evry.fr/wp-content/uploads/2020/07/logo_ibiscsaclay.png
execute:
  echo: true
css: style.css
jupyter: python3
bibliography: references.bib
incremental: true
---


# RNA structure description

## Slides 

Slides are available at:

https://evryrna.github.io/rna_3d_bioinfo_m2_geniomhe/

## Gene expression

![Only 2% of DNA is transcripted into mRNA and then into proteins.](img/transcription_gif.gif){width="120%"}

## RNA roles



::: {.columns}
::: {.column width="30%"}
![](img/4L8R_mRNA.gif){width="100%"}
:::

::: {.column width="70%"}
- RNA biological function is directly linked to its **3D structure**
- RNAs are molecules involved in multiple biological processes, including **diseases**
- RNA 3D structure could be an interest in RNA therapy
- Experimental methods are costly → Need of computational methods
:::
:::

## RNA sequence

![](img/raw_structure.png){width="50%" height="50%" fig-align="center"}

- RNA is composed of 4 different nucleotides
- Linear combination of nucleotides gives its raw sequence


## RNA 2D structure

![](img/all_2d_structures.png){width="100%" fig-align="center"}

- Interactions between complementary base pairs (AU, CG and GU)
- Watson-Crick and Wobble interactions

## RNA 3D structure

![](img/4L8R_mRNA.gif){width="100%" fig-align="center"}

- RNA 3D structure describes the complete folding and spatial arrangement

## RNA structures

![](img/raw_2d_3d.gif)

- 3D structure is constrained by 2D, which is constrained by raw sequence


## Nucleotides 

![Nucleotide description](img/nucleotide_description.png){width="50%"}

## Nucleotides chain

![Adaptation of [article](https://www.genome.gov/genetics-glossary/RNA-Ribonucleic-Acid): description of RNA molecular components](img/rna_schema.drawio.png){width="90%"}


## RNA edges

![RNA edges where H-bonded pairs can be formed](img/rna_edges.drawio.png)

## RNA edges 

![All possible interactions (cis, trans, parallel, antiparallel)](img/rna_all_edges.drawio.png)


## RNA base pairs

- Given orientations, the edges and base pairing, there are more than **200** possible base pairs.

- The standard WC base pair corresponds to the cis WC/WC pairing. 

. . . 

![](img/wc_edge.drawio.png){width="50%" fig-align="center"}

- Non-standard interactions are crucial in RNA structure

## RNA 3D structure

![](img/4L8R_mRNA.gif){fig-align="center"}

RNA 3D structures are complex and involve a lot of different interactions 

# RNA 3D structures prediction

## RNA 3D structures prediction

- Solved structures are stored in the [PDB](https://www.rcsb.org/)

. . .

![](img/pdb_all_rnas.png){width="60%" fig-align="center"}

## RNA 3D structures prediction

![Example of RNA page on PDB](img/pdb_one_rna.png){fig-align="center"}

## RNA 3D structures prediction

![Each RNA 3D structure is the result of hard work](img/pdb_article.png){fig-align="center"}



# Experimental methods 

## Experimental methods 
### Macromolecular X-ray crystallography (MR)

- Most widely used technique
- A crystal is exposed to an X-ray beam. 
- Diffraction images are collected with rotation of the crystal. 
- Application of inverse Fourier to find strucutre

## Experimental methods 
### Nuclear magnetic resonance (NMR)

- Observe local magnetic fields around atom and measure interactions between them
- Atoms carry magnetic that take various orientations characterized by different energies when applied to magnetic field. 

## Experimental methods 

### Electron microscopy (EM)

- Rely on the constrast formed by the interaction of incoming electrons 
- Multiple 2D projection images of randomly oriented molecules are recorded
- 3D density map is computationally combined

## Experimental methods 
![](img/experimental_methods.png){fig-align="center"}

## Experimental methods 

- Experimental methods exist but are laborious and of high cost (in **time** and **money**)

- Computational methods aim to predict RNA 3D structures with high throughput

. . .

**Question**: How to compute RNA 3D structures ? 

# RNA 3D structures prediction

## RNA 3D structures prediction

### General pipeline 

Models that predict RNA 3D structures use this general workflow: 

- Generate RNA 3D candidates 
- Selection of good candidates (structure evaluation)
- Refinement: relaxe the structures 
- Final prediction: return either the best structure or multiple structures

## RNA 3D structures prediction

Methods can be divided into 3 classes: 

- ***Ab initio*** (physics-based): use of thermodynamic and physics
- **Template-based**: use of fragment assembly
- **Deep learning**

## Type of inputs

![Type of inputs usually used to predict RNA 3D structures.](img/timeline_all_papers.drawio.png)

# Ab initio
## Ab initio

- Rely on the fact that native structures are those states with the lowest energies for RNAs
- The energy used usually consider **dihedral angles**, **bond length**, **base pairing**, ...
- Use a **generation** process and a **discriminator**
- Use Monte Carlo or molecular dynamics (MD) sampling procedures


## RNA dihedral angles

![](img/all_angles.drawio.png){width=100% fig-align="center"}



## Ab initio
### Coarse-grained 

![](img/rotation_per_nucloetide.drawio.png){fig-align="center" width="100%"}


- All-atom simulation can be time-consuming
- Use of coarse-grained to reduce computation time 




## Ab initio
### RNAJP granularity

![](img/rnajp_nucleotides.png){fig-align="center" width="40%"}

- 5 pseudoatoms (beads) to represent a nucleotide




## Ab initio
### RNAJP force-field

- Force-field uses 12 terms:
$$
E_{total} = E_{bond} + E_{angle} + E_{torsion} + \\
            E_{base-pairing} + E_{base-stacking} + \\ 
            E_{excluded-volume} + \\
            E_{A helix} + E_{helix-helix}+ E_{junction-compact} + \\
            E_{JAR3D} + E_{loop-loop} + \\
            E_{PK stacking}
$$


## Ab initio
### RNAJP force-field

- $E_{bond} + E_{angle} + E_{torsion}$: bonded interactions in terms of bond length, angle and torsion
- $E_{base-pairing} + E_{base-stacking}$: important base pairing and base stacking interactions
- $E_{excluded-volume}$: avoid steric clashes between beads
- $E_{A helix} + E_{helix-helix}+ E_{junction-compact}$: use to accelerate the conformational sampling


## Ab initio
### RNAJP force-field

- $E_{JAR3D} + E_{loop-loop}$: describe the intraloop and interloop interactions
- $E_{PK stacking}$: describe the stacking interaction between the pseudoknotted (PK) helix and its adjacent non-pseudoknotted (nonPK) helix


## Ab initio
### RNAJP sampling

- Sampling procedure: molecular dynamics (MD) 
- Procedure: 
    - Generate an initial 3D structure
    - Fold the helices in the given 2D structure 
    - Fold the hairpin and internal loop structures 
    - Sample different junction topologies and junction loop structures

## Ab initio
### RNAJP: model selection

- Select the lowest-energy structure as the first predicted model
- Among a pool of top 1% low energy, select the structure with the largest RMSD from previous selected structure


## Ab initio
### RNAJP: model selection

- Select the structure among the top 1% low-energy structures with the largest sum of RMSDs relative to the first and second models as the third model
- Continue until N predictions


## Ab initio
### RNAJP: all-atom reconstruction

- CG beads aligned in the standard base
- Selection of reconstructed atoms after superimposition


## Ab initio
### RNAJP: refinement

- Fix the broken bonds and remove steric clashes using QRNAS (@qrnas)


## Ab initio
### Limits

- Limited by the sequence length (increase computation time) 
- Require efficient sampling function (to avoid useless computation)
- Methods usually slow 


# Template-based

## Template-based

- Creation of 3D structures from known structures (templates) 
- Use of fragment mapping between 2D and 3D
- Align the target sequence to the template
- Structure evaluation to choose the best candidate


## Template-based
### Granularity 

- Looking to template at the nucleotide level is time consuming
- Sample the space not at the atom level but at the level of basic block

## Template-based
### Granularity 
- Basic blocks can be:
    - Three-nucleotides (used in FARNA)
    ![](img/tp_3nt.png){fig-align="center" width="50%"}

## Template-based {transition="none"}
### Granularity 
:::{.nonincremental}
- Basic blocks can be:
    - Three-nucleotides (used in FARNA)
    - Smallest Secondary Elements (SSEs)
:::

![](img/tp_sses.png){fig-align="center" width="50%"}

## Template-based
### Example: 3dRNA

![](img/3drna_workflow.png){fig-align="center"}


## Template-based
### Example: 3dRNA

- Assembled structures are clustered by K-means
- Use RMSD values to do the clustering 
- Centroids of each cluster are evaluated by 3dRNAscore

## Template-based
### Limits

- Rely on representative templates
- Fail to generalise to unseen structures
- Limited by the number of available RNA

# Towards deep-learning

## Towards deep-learning
### Proteins 

- AlphaFold 2(Jumper et al., 2021) has successfully predicted protein 3D structures


![](img/alphafold.png)


## Towards deep-learning
### Proteins 
AlphaFold was not adapted to RNAs:

- RNAs composed of nucleotides (compared to amino acid)
- Disparity in length for RNAs
- Multiple structures possible for RNAs
- Fewer structures available
- AlphaFold is based on Multiple Sequence Alignment (MSA), but not all RNA families are known

## Deep learning

- Recent works try to reproduce AlphaFold
- New release of AlphaFold 3 for DNA, ligand, ion and **RNA**.

![](img/af3_screen.png){fig-align="center"}


## Deep learning 
### Example: DRfold

![](img/preprint_3.png){fig-align="center"}

- Example of work published that had good performances on a recent challenge

## Deep learning 
### DRfold atom representation

![](img/drfold_cg.png){fig-align="center"}

## Deep learning 
### DRfold architecture

![](img/drfold_architecture.png){fig-align="center"}

## Deep learning 
### DRfold embedding

![](img/drfold_emb.png){fig-align="center"}

## Deep learning 
### DRfold transformer

![](img/drfold_transformer.png){fig-align="center"}

## Deep learning 
### DRfold structure

![](img/drfold_structure.png){fig-align="center"}

## Deep learning 
### DRfold geometry

![](img/drfold_geometry.png){fig-align="center"}


## Deep learning
### Rotation/translation

![](img/rot_reconstruction_base.png){fig-align="center"}

- Predict Euclidean transformation from base frame to global position

## Deep learning
### Rotation/translation

![](img/rot_reconstruction.gif){fig-align="center"}

## Deep learning
### Rotation/translation

![](img/cg_and_full.gif){fig-align="center"}

## Deep learning 
### DRfold FAPE

![](img/drfold_fape.png){fig-align="center"}

## Deep learning 
### DRfold loss

$$
L_{e2e} = 1.5 L_{FAPE} + 0.6L_{dist}
$$

with 
$$
L_{dist} = - \sum_{i,j} \sum_{b=1}^{36} y_{ij}^b \log p_{ij}^b
$$

$y_{ij}^b$ assesses if the distance of atom pair $(i, j)$ in the target experimental structure falls into the $b$-th distance

## Deep learning
### DRfold: dataset

- Training set of 3864 unique RNAs extracted from the PDB
- Test set on 40 non-redundant RNAs
- RNA length from 14 to 392 nucleotides

## Deep learning 
### DRfold: results

![](img/drfold_results.png){fig-align="center"}


## Deep learning
### DRfold

- Use of full-atom reconstruction
- Training took 2 weeks on Nvidia A40 GPUwith 32GB of memory
- RNA sequences of 200 nucleotides maximum
- **Doesn't use MSA nor self-distillation**

## Deep learning
### Limits

- Overall performance of deep learning approaches still to be improved
- Lack of sufficient solved structures
- Interpretation of the model
- Integration of environment conditions (ions ?)


# AlphaFold 3

## AlphaFold 2 architecture

![](img/af2_screen.png){fig-align="center"}

- Use attention-based methods with MSA as inputs
- Use of recycling methods
- Predict rotation/translation vectors and torsional angles

## AlphaFold 2 output

![](img/protein_angles.png){width="50%" height="50%" fig-align="center"}

- Torsional angles are specific to proteins and not adaptable to RNAs

## AlphaFold 3

![](img/af3_architecture.png){fig-align="center" }

- New architecture generalisable to different molecules
- Add diffusion methods to replace prediction of rotation/translation vectors

## AlphaFold 3
### Vocabulary

![](img/af2_vs_af3_0.drawio.png)

- AlphaFold 3 has 32 unique tokens compared to 21 for AlphaFold 2 

## AlphaFold 3
### Pairformer



::: {.columns}
::: {.column width="70%"}
![](img/af2_vs_af3_1.drawio.png)
:::

::: {.column width="30%"}
- Reduce the importance of MSA 
- Puts more weight on the pair representation
:::
:::



## AlphaFold 3
### Diffusion

::: {.columns}
::: {.column width="70%"}
![](img/af2_vs_af3_2.drawio.png)
:::

::: {.column width="30%"}
- Remove Gaussian noise from a CG molecule 
- C1' atom per nucleotide for RNA
:::
:::


## AlphaFold 3
### Output

::: {.columns}
::: {.column width="70%"}
![](img/af2_vs_af3_3.drawio.png)
:::

::: {.column width="30%"}
- AF3 outputs a **diffusion loss**, **PAE** (predicted aligned error) and **PDE** (predicted distance error)
:::
:::

## AlphaFold 3
### Architecture

![](img/af3_architecture_0.png)

- Use attention on pair and single representations

## AlphaFold 3
### Architecture

![](img/af3_architecture_1.png)

## AlphaFold 3
### Architecture

![](img/af3_architecture_2.png)


## AlphaFold 3
### Benchmark

![Benchmark of 10 existing methods and AlphaFold 3 on 5 datasets](img/R1107_optimized.gif)


## AlphaFold 3
### Limits

![](img/results_viz.png)

## AlphaFold 3
### Conclusion & limits

- AF3 for RNAs has competitive results 
- AF3 does not generalise to orphan structures. 
- AF3 works well with long sequences (biased by redundant structures)
- Gives more realistic results than state-of-the-art deep learning methods




# RNA quality measurement
## RNA quality measurement

![](img/2d_evaluation_0.drawio.png){width="80%" height="80%" fig-align="center"}

- Can we evaluate the prediction of RNA 2D structure?

## RNA quality measurement

![](img/2d_evaluation_1.drawio.png){width="80%" height="80%" fig-align="center"}

## RNA quality measurement

![](img/2d_evaluation_2.drawio.png){width="80%" height="80%" fig-align="center"}

## RNA quality measurement

![](img/2d_evaluation_3.drawio.png){width="80%" height="80%" fig-align="center"}

- Easy way to count the prediction error

## RNA quality measurement

![](img/rnadvisor_pred.gif){width="60%" height="60%" fig-align="center"}

- Is this a good RNA 3D structure?

## RNA quality measurement

:::: {.columns }

::: {.column width="50%" }

#### Predicted structure
![](img/rnadvisor_pred.gif){width=100% fig-align="center"}
:::

::: {.column width="50%" }

#### Reference structure
![](img/rnadvisor_true.gif){width=100% fig-align="center"}

:::
::::

- Is it now a good structure?

## RNA quality measurement

![](img/af3_pred.gif){width=20% fig-align="center"}

**Given a structure, how to assess its quality ?**

## Metrics
### Definition
:::: {.columns }

::: {.column width="50%" }

#### Predicted structure
![](img/rnadvisor_pred.gif){width=90% fig-align="center"}
:::

::: {.column width="50%" }

#### Reference structure
![](img/rnadvisor_true.gif){width=90% fig-align="center"}

:::
::::

**Metrics**: Quality measurement given a solved structure

## Metrics 

:::: {.columns }

::: {.column width="33%"}

#### General metrics
![Backbone skeleton of an RNA 3D structure](img/general_metrics.gif){width=70% fig-align="center"}

:::

::: {.column width="33%" }

#### Protein-based

![Protein 3D structure](img/protein_structure.gif){width=100% fig-align="center"}

:::

::: {.column width="33%" }

#### RNA-oriented
![RNA torsional angles](img/rna_metrics.png){width=100% fig-align="center"}

:::

- Different types of metrics for different use
::::



## General metrics

![](img/general_metrics.gif){width=70% fig-align="center"}

- **RMSD**: Atom deviation 
- **CLASH** (@molprobity): Steric clashes     
- **$\epsilon$RMSD** (@barnaba): Relative distance

## Metrics 
### Protein-based 

![](img/protein_structure.gif){width="100%" fig-align="center"}

- **GDT-TS** (@gdt): Superimposed residues 
- **TM-score** (@tm-score-rna): Largest deviation 


## Metrics 
### Protein-based 

![](img/protein_structure.gif){width="100%" fig-align="center"}

- **lDDT** (@lddt): Interatomic distance 
- **CAD-score** (@cad-score): Contact-area 

## Metrics 
### RNA-oriented

![](img/rna_metrics.png){width=100% fig-align="center"}

- **INF, DI** (@inf): Key interactions accuracy 
- **P-VALUE** (@p-value): Non-randomness
- **MCQ** (@mcq): Angle deviations

## Metrics
### Example: Interaction Network Fidelity

$$
INF = \sqrt{( \frac{TP}{TP+FP}) \times (\frac{TP}{TP + FN})}
$$
with TP the true positive of the considered interactions 

- When the model reproduces exactly the reference interactions,
|TP| > 1 and FP = FN = 0.  -> The INF value equals 1

## Metrics
### Example: Interaction Network Fidelity
- If none of the interactions of the reference is reproduced in
the predicted model, |TP| = 0, and the INF score equals 0. 

- INF score can be specific to base-pairing interactions ($INF_{bp}$), the base stacking interactions
($INF_{stack}$), or consider both ($INF_{all}$ )


## Metrics

- Metrics help to evaluate a prediction given a solved structure
- No universal metric exists: RMSD is mainly used (but sensible to outliers)

- **Metrics requires having a solved structure, which is not available when model development**


# Scoring functions
## Scoring functions

![](img/scoring_example.png){fig-align="center" width=105%}

 
- When developing models, a lot of structures can be generated and should be ranked
- Ranking can be used to return multiple structures

. . . 

**Scoring function**: predict the quality of a structure **without a known reference**


## Scoring function

:::: {.columns }

::: {.column width="50%"}

#### Knowledge-based
![](img/knowledge_based.png){width=70% fig-align="center"}

:::

::: {.column width="50%" }

#### Deep learning

![](img/deep_scoring.png){width=100% fig-align="center"}

:::

::::



## Scoring function
### Knowledged based

Scoring function based on knowledge-based statistical potentials

- **RASP** (@rasp): Pairwise-distances
<!-- - **$\epsilon$SCORE** (@barnaba): Relative positions -->
- **3dRNASCORE** (@3drnascore): Distance/dihedral angles
- **DFIRE-RNA** (@dfire-rna): Pairwise-distances
- **rsRNASP** (@rsrnasp): Short/long pairwise distances


## Scoring function
### Deep learning based

Scoring functions based on deep learning architectures

- **RNA3DCNN** (@rna3dcnn): Atom grid
- **ARES** (@ares): Atom position/chemical type

## Scoring function

- Scoring function enables the ranking of RNA 3D structures

![](img/scoring_example_0.png){width="100%" fig-align="center"}

## Scoring function

- If we add the native structure in the decoy, would it be well evaluated ?

![](img/scoring_example_1.png){width="100%" fig-align="center"}

## Scoring function

- To evaluate quality of scoring functions, we count the number of times **native structure** is ranked as the **best structure** by the scoring function

![](img/scoring_example_2.png){width="100%" fig-align="center"}

## Scoring function
### Example: RNAD3DCNN

- Predict for each nucleotide an unfitness score
- The unfitness is based on the RMSD of its surroundings and its RMSD 
- A score of 0 means the nucleotide fits into its environment perfectly (native conformation)
- For global quality assessment, the unfitness scores of all nucleotides are accumulated

## Scoring function
### RNAD3DCNN: inputs

- For each nucleotide, a box of 3 channels represents:
    - Atomic occupation number
    - Mass
    - Charge within the grid box
- A box is of size 32x32x32
- The RGB channels are mapped to the three channels

## Scoring function
### RNAD3DCNN: inputs

![](img/rna3dcnn_inputs.png){fig-align="center"}

## Scoring function
### RNAD3DCNN: architecture

![](img/rna3dcnn_architecture.png){fig-align="center"}


## Scoring function
### RNAD3DCNN: dataset

- Use of 619 non-redundant RNAs from NDB website
- Training done on two different datasets:
    - **MD dataset**: a simulated annealing molecular dynamics simulation was done to each RNA 
    - **MC dataset**: fragment assembly method, MC process guided by a low-resolution knowledge-based energy function
- 1 million training samples and 0.2 million validation for each subdataset


## Scoring function
### RNAD3DCNN: results

![Relationship between RMSD and RNA3DCNN for an RNA](img/rna3dcnn_results.png){fig-align="center"}


## Scoring function
### RNAD3DCNN: results

![Number of native structures with minimum scores](img/rna3dcnn_testset.png){fig-align="center"}


## Scoring function
### RNAD3DCNN

- Try to reproduce RMSD with deep learning approaches
- Inspired by CNN image-based models

### Limits

- Deep learning limits: require huge dataset, black box
- Training step biased by the sampling procedures 
- Performances remain limited to novel approaches

## Hands-on

![](img/medium.png){fig-align="center"}

Tutorials available on [Medium](https://medium.com/@clement.bernard.these/rna-3d-structure-in-depth-tutorials-on-how-to-apprehend-this-research-subject-1d138a112f17)

## Questions

- Don't hesitate to contact me if you have any question. 

- Mail address: clement.bernard@univ-evry.fr

![](img/blue_rp18.gif){width="80%" fig-align="center"}


## References

<!-- ## Project proposal

Torsion angle definition
Link between sequence and torsion angle: SPOT-RNA
Torsion angle distribution
Project: predict torsional discretized space from the sequence -->